aea/cli/ipfs_hash.py

```diff
@@ -25,38 +25,86 @@
 This script requires that you have IPFS installed:
 - https://docs.ipfs.io/guides/guides/install/
 """
-import collections
-import csv
+
 import os
-import re
 import sys
 import traceback
 from pathlib import Path
-from typing import Collection, Dict, List, Optional, Tuple, cast
+from typing import Dict, List, Tuple, cast
 
 import click
 
 from aea.cli.utils.constants import HASHES_FILE
-from aea.configurations.base import (
-    AgentConfig,
-    PackageConfiguration,
-    PackageType,
-    _compute_fingerprint,
+from aea.configurations.base import PackageConfiguration, PackageType
+from aea.configurations.constants import (
+    AGENT,
+    CONNECTION,
+    CONTRACT,
+    DEFAULT_AEA_CONFIG_FILE,
+    DEFAULT_CONNECTION_CONFIG_FILE,
+    DEFAULT_CONTRACT_CONFIG_FILE,
+    DEFAULT_PROTOCOL_CONFIG_FILE,
+    DEFAULT_SERVICE_CONFIG_FILE,
+    DEFAULT_SKILL_CONFIG_FILE,
+    PACKAGE_TYPE_TO_CONFIG_FILE,
+    PROTOCOL,
+    SCAFFOLD_PACKAGES,
+    SERVICE,
+    SKILL,
 )
-from aea.configurations.constants import PACKAGE_TYPE_TO_CONFIG_FILE, SCAFFOLD_PACKAGES
+from aea.configurations.data_types import PackageId, PublicId
 from aea.configurations.loader import load_configuration_object
+from aea.helpers.dependency_tree import DependecyTree, dump_yaml, load_yaml, to_plural
+from aea.helpers.fingerprint import check_fingerprint, update_fingerprint
+from aea.helpers.io import from_csv, to_csv
 from aea.helpers.ipfs.base import IPFSHashOnly
 from aea.helpers.yaml_utils import yaml_dump, yaml_dump_all
 
 
+COMPONENT_TO_FILE = {
+    AGENT: DEFAULT_AEA_CONFIG_FILE,
+    SKILL: DEFAULT_SKILL_CONFIG_FILE,
+    CONTRACT: DEFAULT_CONTRACT_CONFIG_FILE,
+    CONNECTION: DEFAULT_CONNECTION_CONFIG_FILE,
+    PROTOCOL: DEFAULT_PROTOCOL_CONFIG_FILE,
+    SERVICE: DEFAULT_SERVICE_CONFIG_FILE,
+}
+
+
 def package_type_and_path(package_path: Path) -> Tuple[PackageType, Path]:
     """Extract the package type from the path."""
     item_type_plural = package_path.parent.name
     item_type_singular = item_type_plural[:-1]
     return PackageType(item_type_singular), package_path
 
 
-def _get_all_packages(packages_dir: Path) -> List[Tuple[PackageType, Path]]:
+def to_package_id(package_path: Path) -> Tuple[PackageId, Path]:
+    """Extract the package type from the path."""
+    item_type_plural = package_path.parent.name
+    item_type_singular = item_type_plural[:-1]
```

Comment:
so what happens when a singular noun already ends with the letter `s` (e.g. physics, species) ?

Reply:
We don't support arbitrary names like that, just the fixed package types